### PR TITLE
IfcActuator properties from Rail

### PR DESCRIPTION
**Pull request by czapplitec**
Removed one redundant property Current and reused the existing one

### CODE REVIEW

File: IFC4x3/Properties/c/ControlPulseCurrent_1NbOptutP7sPdNxheUF3ur/Documentation.md [hunk 1]
```diff
@@ -0,0 +1 @@
+The current of the control pulse of the electric actuator.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/c/ControlPulseCurrent_1NbOptutP7sPdNxheUF3ur/Documentation.md:1]:
```suggestion
The current of the the electric actuator control pulse.
```

File: IFC4x3/Properties/l/LockingForce_0pLtPGaujDkwK16bnNbYZE/Documentation.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Locking force of motor of the point machine.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/l/LockingForce_0pLtPGaujDkwK16bnNbYZE/Documentation.md:1]:
```suggestion
Locking force of the point machine motor.
```

File: IFC4x3/Properties/l/LockingType_1vFkVHKPjAvvxUxFl0WWv7/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="LockingType_1vFkVHKPjAvvxUxFl0WWv7" Name="LockingType" UniqueId="793ee7d1-519b-4ae7-9ede-ecfbc0820e47" PropertyType="p_enumeratedvalue">
```
Comment (SergejMuhic) [inline IFC4x3/Properties/l/LockingType_1vFkVHKPjAvvxUxFl0WWv7/DocProperty.xml:2]: We are missing a primary data type IfcLabel.

File: IFC4x3/Properties/l/LockingType_1vFkVHKPjAvvxUxFl0WWv7/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Indicates whether the locking is inside of the point machine or outside of it.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/l/LockingType_1vFkVHKPjAvvxUxFl0WWv7/Documentation.md:1]:
```suggestion
Indicates whether the locking is inside or outside of the point machine.
```

File: IFC4x3/Properties/m/MaximumOperatingTime_31nhczYx96JuYbkJ32Ejs4/Documentation.md [hunk 5]
```diff
@@ -0,0 +1 @@
+The maximum time the turnout has to complete the movement before the interlocking turns to out of control status.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/m/MaximumOperatingTime_31nhczYx96JuYbkJ32Ejs4/Documentation.md:1]:
```suggestion
The maximum time in which the turnout has to complete the movement before the interlocking turns to out of control status.
```

File: IFC4x3/Properties/m/MinimumOperatingSpeed_0OpoQ0$Az5huboLE7511z3/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+Minimum operation speed of point machine.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/m/MinimumOperatingSpeed_0OpoQ0$Az5huboLE7511z3/Documentation.md:1]:
```suggestion
Minimum point machine operation speed.
```

File: IFC4x3/Properties/n/NumberOfTractionPoints_1Nk$_0PhT2kBz$hK6Rfjvq/Documentation.md [hunk 7]
```diff
@@ -0,0 +1 @@
+The number of leverages or actuatrs needed to move turnouts.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/n/NumberOfTractionPoints_1Nk$_0PhT2kBz$hK6Rfjvq/Documentation.md:1]:
```suggestion
The number of leverages or actuators needed to move turnouts.
```

File: IFC4x3/Properties/p/PointType_0gfdQdlSPCaRPYyjTFQR6P/DocProperty.xml [hunk 8]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PointType_0gfdQdlSPCaRPYyjTFQR6P" Name="PointType" UniqueId="2aa676a7-bdc6-4c91-b662-f2d74f69b199" PropertyType="p_enumeratedvalue">
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PointType_0gfdQdlSPCaRPYyjTFQR6P/DocProperty.xml:2]: Primary datatype IfcLabel missing.

File: IFC4x3/Properties/p/PointType_0gfdQdlSPCaRPYyjTFQR6P/Documentation.md [hunk 9]
```diff
@@ -0,0 +1 @@
+Type of the railway point，different types of point allow trains to pass at different speeds。
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PointType_0gfdQdlSPCaRPYyjTFQR6P/Documentation.md:1]:
```suggestion
Type of the railway point. Different point types allow trains to pass at different speeds。
```

File: IFC4x3/PropertyConstants/e/ELECTRO-HYDRAULICPOINTMACHINE_3hTKC6DsX7gQm5PhvlvvvI.xml [hunk 10]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ELECTROHYDRAULICPOINTMACHINE_3hTKC6DsX7gQm5PhvlvvvI" Name="ELECTRO-HYDRAULICPOINTMACHINE" UniqueId="eb754306-3768-47a9-ac05-66be6fe79e52" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/e/ELECTRO-HYDRAULICPOINTMACHINE_3hTKC6DsX7gQm5PhvlvvvI.xml:2]:
```suggestion
<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ELECTROHYDRAULICPOINTMACHINE_3hTKC6DsX7gQm5PhvlvvvI" Name="ELECTRO_HYDRAULICPOINTMACHINE" UniqueId="eb754306-3768-47a9-ac05-66be6fe79e52" />
```

File: IFC4x3/PropertyConstants/e/ELECTRO-PNEUMATICPOINTMACHINE_0arRgEzsn1ih$cTMDM7iBB.xml [hunk 11]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ELECTROPNEUMATICPOINTMACHINE_0arRgEzsn1ihcTMDM7iBB" Name="ELECTRO-PNEUMATICPOINTMACHINE" UniqueId="24d5ba8e-f76c-41b2-bfe6-7563561ec2cb" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/e/ELECTRO-PNEUMATICPOINTMACHINE_0arRgEzsn1ih$cTMDM7iBB.xml:2]:
```suggestion
<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="ELECTROPNEUMATICPOINTMACHINE_0arRgEzsn1ihcTMDM7iBB" Name="ELECTRO_PNEUMATICPOINTMACHINE" UniqueId="24d5ba8e-f76c-41b2-bfe6-7563561ec2cb" />
```

File: IFC4x3/PropertyConstants/e/EXTERNAL_3gdw0DUw5Dnhn8UTc5nQnZ.xml [hunk 12]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="EXTERNAL_3gdw0DUw5Dnhn8UTc5nQnZ" Name="EXTERNAL" UniqueId="ea9fa00d-7ba1-4dc6-bc48-79d985c5ac63" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/e/EXTERNAL_3gdw0DUw5Dnhn8UTc5nQnZ.xml:2]: There are a couple of externals already. Would we consider re-using?

File: IFC4x3/PropertyConstants/i/INTERNAL_14$SxyqYLC0ve$1c4mfX0q.xml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyConstant xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="INTERNAL_14SxyqYLC0ve1c4mfX0q" Name="INTERNAL" UniqueId="44fdcefc-d225-4c03-9a3f-066130a61034" />
```
Comment (SergejMuhic) [inline IFC4x3/PropertyConstants/i/INTERNAL_14$SxyqYLC0ve$1c4mfX0q.xml:2]: Same as with external.

File: IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRail/PropertySets/Pset_PointMachine/Documentation.md [hunk 14]
```diff
@@ -0,0 +1 @@
+Properties of point machine used in railway. The property set can be used by different predefined types of IfcActuator, indicated that such actuator is a point machine that can switch and lock the track turnout.
```
Comment (SergejMuhic) [inline IFC4x3/Sections/Domain specific data schemas/Schemas/IfcRail/PropertySets/Pset_PointMachine/Documentation.md:1]:
```suggestion
Properties of a point machine used in railway. The property set can be used by different predefined types of IfcActuator, indicated that such an actuator is a point machine that can switch and lock the track turnout.
```

**Review by SergejMuhic — CHANGES_REQUESTED** [review]
A few suggestions that would make the descriptions a bit more understandable, couple of changes required for constant names and a couple of suggestions to use existing constants.

Also, primary datatype on enumeration properties is missing.

File: IFC4x3/Properties/l/LockingType_1vFkVHKPjAvvxUxFl0WWv7/DocProperty.xml (hunk 3 above)
Comment (jmirtsch) [reply IFC4x3/Properties/l/LockingType_1vFkVHKPjAvvxUxFl0WWv7/DocProperty.xml:2]: I don't believe IfcDoc enforced explicit serialization of an "IfcLabel" datatype on a property template, this is the default.
You can see that here, https://github.com/buildingSMART/IfcDoc/blob/0a884dd67a2d0025952467b956f34b0b2b5cae74/Formats/IfcDoc.Formats/FormatHTM.cs#L2527

Although I see that isn't happening when publishing properties to ifcxml etc.  Is it desired to have it explicit?
There is an impact (not measured how significant) on the serialization size and efficiency if all attributes are explicit.  There are other aspects that "default" (I need to check to find other examples).

File: IFC4x3/Properties/c/ControlPulseCurrent_1NbOptutP7sPdNxheUF3ur/Documentation.md [hunk 15]
```diff
@@ -0,0 +1 @@
+The current of the the electric actuator control pulse.
```
Comment (pjanck) [inline IFC4x3/Properties/c/ControlPulseCurrent_1NbOptutP7sPdNxheUF3ur/Documentation.md:1]: Typo: "the the"

File: IFC4x3/Properties/i/InstallationLimit_0_Xq_kDdTAXR3cFcN9ZofG/Documentation.md [hunk 16]
```diff
@@ -0,0 +1 @@
+Installation limit of the device.
```
Comment (pjanck) [inline IFC4x3/Properties/i/InstallationLimit_0_Xq_kDdTAXR3cFcN9ZofG/Documentation.md:1]: Limit in what sense? I'd love a bit more explanation here.

File: IFC4x3/PropertyConstants/e/EXTERNAL_3gdw0DUw5Dnhn8UTc5nQnZ.xml (hunk 12 above)
Comment (pjanck) [reply IFC4x3/PropertyConstants/e/EXTERNAL_3gdw0DUw5Dnhn8UTc5nQnZ.xml:2]: +1

File: IFC4x3/Properties/p/PointType_0gfdQdlSPCaRPYyjTFQR6P/Documentation.md [hunk 17]
```diff
@@ -0,0 +1 @@
+Type of the railway point. Different point types allow trains to pass at different speeds.
```
Comment (pjanck) [inline IFC4x3/Properties/p/PointType_0gfdQdlSPCaRPYyjTFQR6P/Documentation.md:1]: Is it clear what `point` refers to here?

File: IFC4x3/PropertyEnumerations/l/PEnum_LockingType .xml [hunk 18]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocPropertyEnumeration xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PEnum_LockingType" Name="PEnum_LockingType " UniqueId="a5ea0ccf-3d9f-4bad-8779-8be431d18183">
+	<Constants>
+		<DocPropertyConstant xsi:nil="true" href="EXTERNAL_3gdw0DUw5Dnhn8UTc5nQnZ" />
+		<DocPropertyConstant xsi:nil="true" href="INTERNAL_14SxyqYLC0ve1c4mfX0q" />
+	</Constants>
+</DocPropertyEnumeration>
+
```
Comment (pjanck) [inline IFC4x3/PropertyEnumerations/l/PEnum_LockingType .xml:8]: There's a space in the filename.

**Review by pjanck — CHANGES_REQUESTED** [review]
Small mistakes found. Must be addressed before merge.

**Pull request closed without merging**